File: internal/util/path.go
package util

import (
	"os"
	"path/filepath"
	"strings"
)

// ExpandHome expands "~/" to the user's home directory if possible.
func ExpandHome(path string) string {
	if path == "" {
		return path
	}
	if strings.HasPrefix(path, "~/") || path == "~" {
		home, err := os.UserHomeDir()
		if err != nil {
			return path
		}
		if path == "~" {
			return home
		}
		return filepath.Join(home, path[2:])
	}
	return path
}

File: internal/util/path.ts
import { homedir } from "node:os";
import path from "node:path";

export function expandHome(value: string): string {
  if (!value) return value;
  if (value === "~") return homedir();
  if (value.startsWith("~/")) {
    return path.join(homedir(), value.slice(2));
  }
  return value;
}
